File: go/event_client.go
package synapse

import (
	"github.com/streadway/amqp"
	"log"
	"github.com/bitly/go-simplejson"
)

/**
发送一个事件
 */
func (s *Server) SendEvent(eventName string, params map[string]interface{}) {
	if s.DisableEventClient {
		log.Printf("[Synapse Error] %s: %s \n", "Event Send Not Success", "DisableEventClient set true")
	} else {
		query := simplejson.New();
		query.Set("from", s.AppName)
		query.Set("to", "event")
		query.Set("action", eventName)
		query.Set("params", params)
		queryJSON, _ := query.MarshalJSON()
		err := s.mqch.Publish(
			s.SysName, // exchange
			"event." + s.AppName + "." + eventName, // routing key
			false, // mandatory
			false, // immediate
			amqp.Publishing{
				ContentType: "application/json",
				CorrelationId: s.randomString(20),
				Body:        []byte(queryJSON),
			})
		s.failOnError(err, "Failed to publish a event")
		if s.Debug {
			log.Printf("[Synapse Debug] Publish Event: %s.%s %s", s.AppName, eventName, queryJSON)
		}

	}
}

File: python/event_client.py
# coding=utf-8
import uuid,pika,json
from .base import Base

class EventClient(Base):
    @classmethod
    def send_event(self,action,params):
        if self.disable_event_client:
            self.log("[Synapse Error] Event Send Not Success: DisableEventClient set true")
        else:
            data = {
                "from": self.app_name,
                "to" : "event",
                "action": action,
                "params": params
            }
            self.mqch.basic_publish(exchange=self.sys_name,
                                    routing_key="event."+self.app_name+"."+action,
                                    properties=pika.BasicProperties(
                                            correlation_id = str(uuid.uuid4()),
                                            ),
                                    body=json.dumps(data))
            if self.debug:
                    self.log("[Synapse Debug] Publish Event: %s.%s %s" % (self.app_name,action,data))